impl/src/test/java/slack/SlackResourceImplTest.java

```diff
@@ -6,106 +6,243 @@
 import com.github.seratch.jslack.api.methods.response.users.UsersInfoResponse;
 import com.github.seratch.jslack.api.model.Message;
 import com.github.seratch.jslack.api.model.User;
+import com.github.seratch.jslack.api.model.block.DividerBlock;
+import com.github.seratch.jslack.api.model.block.LayoutBlock;
+import org.junit.Before;
 import org.junit.Test;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.UUID;
 
+import static java.util.Arrays.asList;
 import static java.util.Collections.singletonList;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static rx.Observable.just;
+import static slack.SlackResourceImpl.COULD_NOT_FETCH_USER_EMAIL_FROM_SLACK;
+import static slack.SlackResourceImpl.COULD_NOT_GET_USER_BY_EMAIL_FROM_SLACK;
+import static slack.SlackResourceImpl.USER_EMAIL_FROM_SLACK_IS_NULL_AND_MIGHT_BE_DUE_TO_MISSING_SCOPE_USERS_READ_EMAIL;
 
 public class SlackResourceImplTest {
 
-    @Test
-    public void handleUserInfoResponse() {
-        UsersInfoResponse usersInfoResponse = new UsersInfoResponse();
-        usersInfoResponse.setOk(true);
+    private SlackClient slackClient;
+    private SlackResource slackResource;
+    private SlackConfig slackConfig;
+
+    @Before
+    public void shouldDoStuff() {
+        slackConfig = new SlackConfig();
+        slackConfig.setApiToken("api-token");
+        slackConfig.setBotUserToken("bot-token");
+        slackClient = mock(SlackClient.class);
+        slackResource = new SlackResourceImpl(slackConfig, slackClient);
+    }
 
-        final User.Profile profile = new User.Profile();
-        profile.setEmail("tst@a.b");
+    @Test
+    public void shouldReturnUserEmailBySlackId() {
+        // given that slack has the user we are asking for
+        UsersInfoResponse usersInfoResponse = createSuccessfulUserResponse();
+        final String slackId = usersInfoResponse.getUser().getId();
```

Comment:
Does slackId matter here, since we're not asserting the contents of the request to slackClient?
Same for the test cases below.

Reply:
ah its the userId , will rename :)